extensions/notebook/src/jupyter/jupyterNotebookManager.ts

```diff
@@ -46,7 +46,7 @@ export class JupyterNotebookManager implements nb.NotebookManager, vscode.Dispos
 	}
 
 	dispose() {
-		if (this._sessionManager) {
+		if (this._sessionManager && this.sessionManager.isReady) {
```

Comment:
For this one, can we fix in the `shutdownAll` method instead? That's what we do for regular shutdown where if not ready we auto-resolve the shutdown and don't call through to the APIs.

Reply:
@kevcunnane, I tried this. If removed this check and check isReday in JuptyerSessionManager, will hit The issue will happen to  this._sessionManager.dispose() in then.

[Microsoft.notebook]Cannot read property 'dispose' of undefined
console.(anonymous function) @ D:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js:2280
MainThreadExtensionService.$onExtensionRuntimeError @ mainThreadExtensionService.ts:42
RPCProtocol._doInvokeHandler @ rpcProtocol.ts:461
RPCProtocol._invokeHandler @ rpcProtocol.ts:446
RPCProtocol._receiveRequest @ rpcProtocol.ts:366
RPCProtocol._receiveOneMessage @ rpcProtocol.ts:296
(anonymous) @ rpcProtocol.ts:171
Emitter.fire @ event.ts:228
acceptChunk @ ipc.net.ts:103
Protocol._socketDataListener @ ipc.net.ts:132
ZoneDelegate.invokeTask @ D:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js:421
Zone.runTask @ D:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js:188
ZoneTask.invokeTask @ D:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js:496
ZoneTask.invoke @ D:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js:485
emitOne @ events.js:116
emit @ events.js:211
addChunk @ _stream_readable.js:263
readableAddChunk @ _stream_readable.js:250
Readable.push @ _stream_readable.js:208
onread @ net.js:594
D:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js:2280 TypeError: Cannot read property 'dispose' of undefined
	at JupyterSessionManager.dispose